frontend/app/src/components/accounts/ExchangeBalances.vue

```diff
@@ -0,0 +1,272 @@
+                </span>
+              </div>
+            </template>
+          </v-select>
+        </v-col>
+        <v-col cols="2" class="hidden-sm-and-down">
+          <v-tabs
+            fixed-tabs
+            vertical
+            hide-slider
+            optional
+            class="exchange-balances__tabs"
+          >
+            <v-tab
+              v-for="(exchange, i) in connectedExchanges"
+              :key="i"
+              class="exchange-balances__tab ml-3 my-3 py-3"
+              active-class="exchange-balances__tab--active"
+              :to="`/accounts-balances/exchange-balances/${exchange}`"
+              @click="selectedExchange = exchange"
+            >
+              <v-img
+                contain
+                height="48"
+                width="48"
+                :title="exchange"
+                :src="require(`@/assets/images/${exchange}.png`)"
+                class="exchange-balances__tab__icon"
+              />
+              <div class="exchange-balances__tab__title d-block mt-2">
+                {{ exchange }}
+              </div>
+              <div class="exchange-balances__tab__amount d-block">
+                <amount-display
+                  show-currency="symbol"
+                  fiat-currency="USD"
+                  :value="exchangeBalance(exchange)"
+                ></amount-display>
+              </div>
+            </v-tab>
+          </v-tabs>
+        </v-col>
+        <v-col>
+          <asset-balances
+            v-if="exchange"
+            :balances="exchangeBalances(exchange)"
+          ></asset-balances>
+          <div v-else class="pa-4">
+            Select an exchange to view asset details.
+          </div>
+        </v-col>
+      </v-row>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { default as BigNumber } from 'bignumber.js';
+import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
+import { createNamespacedHelpers } from 'vuex';
+import AmountDisplay from '@/components/display/AmountDisplay.vue';
+import AssetBalances from '@/components/settings/AssetBalances.vue';
+import { AssetBalance } from '@/model/blockchain-balances';
+import { ExchangeInfo } from '@/typing/types';
+import { Zero } from '@/utils/bignumbers';
+
+const { mapState, mapGetters } = createNamespacedHelpers('balances');
+
+@Component({
+  components: {
+    AssetBalances,
+    AmountDisplay
+  },
+  computed: {
+    ...mapState(['connectedExchanges']),
+    ...mapGetters(['exchangeBalances', 'exchanges'])
+  }
+})
+export default class ExchangeBalances extends Vue {
+  @Prop({ required: false, default: '' })
+  exchange!: string;
+  selectedExchange: string = '';
+
+  connectedExchanges!: string[];
+  exchanges!: ExchangeInfo;
+  exchangeBalances!: (exchange: string) => AssetBalance[];
+
+  mounted() {
+    this.selectedExchange = this.selectedExchange = this.$route.params.exchange;
+  }
+
+  @Watch('$route')
+  onRouteChange() {
```

Comment:
As an alternative you could use the route guards for the same thing
https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards

Reply:
I think I will leave as-is for now unless you have a strong objection to this?

Reply:
no strong objection I can change it in the future :+1: